src/engine/source/base/result.hpp

```diff
@@ -0,0 +1,115 @@
+#ifndef _RESULT_H
+#define _RESULT_H
+
+#include <string>
+
+namespace base::result
+{
+
+template<typename Event>
+class Result
+{
+private:
+    Event m_payload;
+    std::string m_trace;
+    bool m_success;
+
+public:
+    Result(Event payload, std::string trace, bool success)
+        : m_payload {payload}
+        , m_trace {trace}
+        , m_success {success}
+    {
+    }
+
+    Result(const Result& other)
+        : m_payload {other.m_payload}
+        , m_trace {other.m_trace}
+        , m_success {other.m_success}
+    {
+    }
+
+    Result(Result&& other)
+        : m_payload {std::move(other.m_payload)}
+        , m_trace {other.m_trace}
+        , m_success {other.m_success}
+    {
+    }
+
+    Result& operator=(const Result& other)
+    {
+        m_payload = other.m_payload;
+        m_trace = other.m_trace;
+        m_success = other.m_success;
+        return *this;
+    }
+/*
+    Result& operator=(Result&& other)
+    {
+        m_payload = std::move(other.m_payload);
+        m_trace = other.m_trace;
+        m_success = other.m_success;
+        return *this;
+    }
+*/
```

Comment:
Apply the rule ~of three or rule~ of five

Reply:
Done